[PIE-2] Feature/Dough Staking UI

TotoFullStackDev opened the pull request:
##### Description
<!-- A description on what this PR aims to solve. -->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Test coverage hasn't decreased

##### Testing
<!-- Why should the PR reviewer trust that this change doesn't break
  anything? How have you tested this change?
-->

##### Refers/Fixes
<!--
  Link to an issue if applicable. For example:
  If your PR fixes an issue -- Fixes: #102
  If your PR refers an issue -- Refs: #101
-->

Commit:
first sketch for new dough staking page

Commit:
added contract address, routes, and abis

Commit:
added return/resolve into approve async function

Commit:
updated doughStaking address

Commit:
added missing dough address

Commit:
added overrides on approve function

Commit:
fixed staking contract address, owner bug fixed on smart contract

Alexintosh committed:
Adding sanity checks and reactivity to input responce

Alexintosh committed:
Merkle root verification

Alexintosh committed:
Claim reward token

Alexintosh committed:
New address

Commit:
changed contract addresses for new ones

nicoampx committed:
token icons

nicoampx committed:
layout

nicoampx committed:
buttons

nicoampx committed:
card title

nicoampx committed:
summary

nicoampx committed:
paddings

nicoampx committed:
icons

nicoampx committed:
your staking

nicoampx committed:
your staking

nicoampx committed:
disabled position

nicoampx committed:
typos

nicoampx committed:
past rewards

nicoampx committed:
mobile

Commit:
first implementation of dymamic data from smart contract, to be completed

Commit:
added new contract's addresses into json file

Commit:
fixed eslint errors

Commit:
fixed errors on MerkleTree after eslint updates

Commit:
added boostToMax action

Commit:
fixed 3-years staking button

Commit:
implemented subgraph for all staking infos, still to be deeply tested

Commit:
small cleanup

Commit:
added check on claim button

Commit:
added ejected condition on UI

Commit:
fixed boostToMax button

Commit:
fixed boostToMax button

Commit:
fixed boostToMax button

Commit:
added compatibility between graph and contract response

Commit:
fixed bug, added claimed blocks

Alexintosh committed:
Avoiding the page to break on cycling on an undefined array

Alexintosh committed:
Fixing little things

Alexintosh commented:
## Notes

- Cursor pointer on buttons
- Boost to max is a big deal, we need to celebrate that with cool animations!
- accountWithdrawableRewards coming from the graph is always 0

Commit:
temp fix for unstake testing

Commit:
added pointers to buttons

Commit:
removed distributed rewards

Commit:
Only using the graph to fill the past rewards yellow box, for anything else use on chain data

Commit:
PIE-63.1 - Change veDough logo for Dough

Commit:
PIE-63.2 - Add veDOUGH accrued

Commit:
added dough and vedough to total staked box

Commit:
PIE-63.5 - Format all money-field properly

Commit:
PIE-66.4 - Transaction link does not work on all cases

Commit:
PIE-66.5 - Add a call-out/mouse over function to explain why rewards were slashed

Commit:
PIE-64.2 - Decimals allowed on month selection

Commit:
PIE-65 completed, fixed eslint error

Commit:
fixed eslint error

Commit:
PIE-63 - hide duplicated boosted items, and disabled boosted button on already-boosted-items

Commit:
PIE-142 - BoostToMax should not show on locks of 36 months

Commit:
PIE-143 - Default sorting by lock timestamp desc on Your Staking box

Commit:
PIE-141 - Unstake can be clicked even when returns an error

Commit:
PIE-141 - added status to locks

Commit:
PIE-22 - just to link jira with github

Alexintosh reviewed (changes requested):
linter is not passing

TotoFullStackDev commented:
> linter is not passing

i'm on it

Commit:
fixed eslint errorr

TotoFullStackDev committed:
Merge branch 'development' into feature/dough-staking-UI

Commit:
PIE-22: just a commit to fix jira warnings

Commit:
PIE-205: bug fixed

Commit:
typo fixed

Commit:
fixed PIE-201 and PIE-202, icons still missing

nicoampx committed:
emojis for topmenu

nicoampx committed:
3 years button animation

gradient and shaking to attract attention

nicoampx committed:
error message balance for insufficient balance

Commit:
reverted sorting on locks

Commit:
PIE-126 - fixed point 3

Commit:
PIE-126 - fixed point 4

Commit:
PIE-126 - fixed point 5

Commit:
added approving label on button

Commit:
fixed bugs for PIE-126

Commit:
fixed PIE-216 and PIE-224

Commit:
PIE-158 - added 2 new pages

Commit:
PIE-158 added staking helper, still to be improved

Commit:
improved staking helper

Commit:
completed rewards and positions, just staking page is missing a restyle

Commit:
added async load and events on new components

Commit:
fixed bug on UI reactive updates

Commit:
exported stakeDOUGH function into staking-helper

Commit:
created stakingSummary component, and exported claim function into the stakingHelper

Commit:
improved stakingSummary component, made it self-reactive when stakingData changes

Commit:
exported approvetoken function into stakingHelper

nicoampx committed:
fade-in animation

Commit:
fixed approve bug

Commit:
completed PIE-321

TotoFullStackDev committed:
Merge branch 'development' into feature/dough-staking-UI

Commit:
completed PIE-237

Commit:
fixed eslint errors on staking helper

nicoampx committed:
replacing vedough token image urls

the url changed because of moving the images to its relevant folder on the brand repo

Commit:
fixed veDOUGH broken icon

Commit:
PIE-261 - changed url names

Commit:
PIE-261 - changed url names

Commit:
PIE-261 - changed titles

Commit:
PIE-233 - completed restyle, and aligned bottom infos

Commit:
PIE-247 - completed

Commit:
added Adrian's eth address as valid participant

Commit:
reverted participations json file

Commit:
setup stakingEnds in minute, for testing purposes

Commit:
fixed bug after latest commit

Commit:
PIE-373 - bug fixed

Commit:
PIE-373 - added unsubscribe when reloading

Commit:
PIE-381 - bug fixed

Commit:
PIE-380 - bug fixed

Commit:
PIE-380 - fixed other bug

Commit:
PIE-380 - fixed other bug

Commit:
fixed 3-years button bug

Commit:
PIE-390 - bug fixed

Commit:
PIE-392 - bug fixed

Commit:
PIE-397 - bug fixed

nicoampx committed:
fixing icon spacing

nicoampx committed:
text alignment

nicoampx committed:
dough staked simulator page

reference tot dough staked on simulator page

nicoampx committed:
DOUGH staked token page

Reference DOUGH staked on the DOUGH token page

Commit:
PIE-391 - bug fixed

Commit:
PIE-395 - added visual loader

Commit:
PIE-394 - visual improvment

Commit:
PIE-388 - improvement completed

Commit:
PIE-234 - improvement completed

Commit:
PIE-396 - improvement completed

Commit:
PIE-396 - fixed wrong color

Commit:
PIE-404 - first approach

nicoampx committed:
fixing css

Commit:
PIE-404 - completed summary box

Commit:
PIE-404 - fixed small bug on zero-out values

Commit:
PIE-404 - completed component, css still missing

Commit:
PIE-404 - completed component with progress/percentage bars

nicoampx committed:
key stats css

Commit:
PIE-404 - task completed

Commit:
PIE-403 - restake button behavior

nicoampx committed:
header style improvements

- logo does not stretch anymore
- responsive behaviour that makes some element disappear in smaller screens
- padding and margins

nicoampx committed:
Merge branch 'feature/dough-staking-UI' of https://github.com/pie-dao/pools.piedao.org into feature/dough-staking-UI

nicoampx committed:
small css fixes

nicoampx committed:
page padding css

nicoampx committed:
modal paddings

nicoampx committed:
placeholder disappears on focus

nicoampx committed:
lockin period input to type=number

so use can use keyboard arrows

Commit:
PIE-250 - boosted success modal

Commit:
PIE-250 - boosted success modal

Commit:
PIE-250 - bug fixed

Commit:
PIE-250 - more bugs fixed

Commit:
enabled back boostToMax

Commit:
PIE-404 - completed

Commit:
PIE-395 - task completed

Commit:
PIE-411, PIE-395, fixed bug on logout for visual loader indicator

Commit:
PIE-250 - moved boosted modal into a component

Commit:
PIE-250 - fixed modal opening time

Commit:
PIE-251 - just CTA are missing

Commit:
added flag to show/hide loader on stats component

Commit:
PIE-251 - enabled staking back

Commit:
reported fixes for mainnet on testnet branch

Commit:
removed hardcoded value of 60, replaced with global const

Commit:
removed hardcoded value of 60, replaced with global const

Commit:
PIE-416 - bug fixed

Commit:
fixed bug on average locks duration

Commit:
PIE-420 - bug fixed

Commit:
PIE-248 - added metamask button

Commit:
fixed amount fetching from balances

Commit:
fixed bug on voting power and lock average

Commit:
cleanup for incoherent variables

nicoampx committed:
loader css fixes

nicoampx committed:
Merge branch 'feature/dough-staking-UI' of https://github.com/pie-dao/pools.piedao.org into feature/dough-staking-UI

nicoampx committed:
deleting deprecated component totdoughstaked

nicoampx committed:
deleting deprecated component totdoughstaked from the page

nicoampx committed:
keystats container

and deleting deprecated component totdoughstaked

Commit:
aligned frontend to graph changes

Commit:
PIE-423 - bug fixed

Commit:
PIE-425: bug fixed

Commit:
PIE-426 - bug fixed

Commit:
PIE-429 - bug fixed

Commit:
PIE-424 and PIE-428 - bug fixed

Commit:
PIE-417 - bug fixed

Commit:
removed useless consolelog

Commit:
PIE-427 - fixed layout

Commit:
PIE-251 - stake success modal completed

Commit:
PIE-251 - fixed bug on animations and data updates

Commit:
PIE-434 - bug fixed

Commit:
PIE-437 - bug fixed

Commit:
added dynamic account switcher, in sync with metamask

Commit:
added dynamic account switcher, in sync with metamask

Commit:
PIE-442 - bug fixed

Commit:
PIE-251 - reduced animation interval

Commit:
PIE-251 - extended clickable are to whole div

Commit:
PIE-251 - fixed legend-case bug

nicoampx committed:
hero text and benefit section

nicoampx committed:
gov model and key stats

nicoampx committed:
doughconomics section

nicoampx committed:
snapshot and we are hiring section

Commit:
duplicated dough page for Nico

nicoampx committed:
dough staking campaing page

Commit:
added snapshot graphql

Commit:
Merge branch 'feature/dough-staking-UI' of https://github.com/pie-dao/pools.piedao.org into feature/dough-staking-UI

Commit:
added countdown

Commit:
aligned routes

TotoFullStackDev committed:
Merge branch 'development' into feature/dough-staking-UI

Commit:
changed PIP-XX for PIP-60 on staking page

nicoampx committed:
content and counter

nicoampx committed:
Merge branch 'feature/dough-staking-UI' of https://github.com/pie-dao/pools.piedao.org into feature/dough-staking-UI

Commit:
fixed eslint errors
